### ISSUE TITLE
Update

### ISSUE THREAD

**Issue by KilluaGremory** [Yink/Amadeus#110]
Will there be any new app updates?

**Comment by Ibnelaiq** [comment]
Very low posibility. You can try asking same question in his [discord channel](https://discord.com/channels/213420119034953729/440507759604137984)

**Comment by icpryde** [comment]
How cool would it be to add chat gpt with this acting like Kurisu it would actually make the app useful but also stay in the spirit of the app itself

**Comment by Ibnelaiq** [comment]
Yeah, we can prepare chatgpt like this: https://chat.openai.com/share/8f825dac-d02f-43bc-87ab-289a2e06c17c

**Comment by AkechiShiro** [comment]
@KilluaGremory I'll be trying to update the app but I can't tell you if I'll be successful, I'm total newb to Javascript and Android application dev.

I'd like to link it to Gpt4o so we can get Kurisu's response in under 300ms

**Comment by icpryde** [comment]
> @KilluaGremory I'll be trying to update the app but I can't tell you if I'll be successful, I'm total newb to Javascript and Android application dev.
> 
> I'd like to link it to Gpt4o so we can get Kurisu's response in under 300ms

Oh man that would be amazing! It’s so crazy how we literally now have the technology to pull off real Kurdish from Amadeus now! Definitely keep us up to date

**Comment by KilluaGremory** [comment]
> @KilluaGremory I'll be trying to update the app but I can't tell you if I'll be successful, I'm total newb to Javascript and Android application dev.
> 
> I'd like to link it to Gpt4o so we can get Kurisu's response in under 300ms

Yes, I would also like it to connect to chatgpt4o

**Comment by Ibnelaiq** [comment]
@KilluaGremory @icpryde @AkechiShiro 
I found it suitable to mention Chatgpt4o API pricing of $5.00 / 1M tokens (Input) & $15.00 / 1M tokens (Output) [1].
We can also check how many tokens are generated on average in a sentence by this tool https://platform.openai.com/tokenizer 
Example: "Hi, what is your name" contains 6 tokens [2].
If we can create a mutual payment system like all users, We can update this app or create a new one. 

[1] = [Pricing chatgpt4o](https://openai.com/api/pricing/#:~:text=a%2050%25%20discount.-,GPT%2D4o,-GPT%2D4o%20is)
[2] = [Chatgpt 3 tokenizer](https://imgur.com/a/JGe2OIP)

**Comment by AkechiShiro** [comment]
@Ibnelaiq we would still need a platform for this mutual payment system or some kind of system that would allow that. 

For the dev part, here is the plan : 
- Update to the latest dependencies/squash all known bugs or rewrite the app in native Flutter (Android and iOS support)
- Integrate ChatGPT so that Kurisu can access knowledge from October 2023 in the whole World.
- Attempt to integrate GPT4o using most likely paid credits, I will need to pitch in some money, maybe once I get at that step crowdfunding the credits would be a good idea?  
- Finally, I do not know if there is TTS for Kurisu's voice in the app right now, meaning I might need to work on that using all the sounds from Kurisu that are available.

**Comment by AkechiShiro** [comment]
@Ibnelaiq I'll be creating a Discord for people who want to help, test and suggest, would you be okay to be come a moderator there ?

**Comment by icpryde** [comment]
@AkechiShiro not 100% sure how it works but it could just set it up so we use our own personal api from ChatGPT with the software? I imagine that would be easier?

**Comment by KilluaGremory** [comment]
> @Ibnelaiq we would still need a platform for this mutual payment system or some kind of system that would allow that. 
> 
> For the dev part, here is the plan : 
> - Update to the latest dependencies/squash all known bugs or rewrite the app in native Flutter (Android and iOS support)
> - Integrate ChatGPT so that Kurisu can access knowledge from October 2023 in the whole World.
> - Attempt to integrate GPT4o using most likely paid credits, I will need to pitch in some money, maybe once I get at that step crowdfunding the credits would be a good idea?  
> - Finally, I do not know if there is TTS for Kurisu's voice in the app right now, meaning I might need to work on that using all the sounds from Kurisu that are available.

Shouldn't we ask Yink's permission first to do something like that?

**Comment by icpryde** [comment]
Also link the discord I’d be happy to join in I imagine we can use ChatGPT to help with the coding as well

**Comment by AkechiShiro** [comment]
@icpryde yes but you would need to pay for the APIuse if you use GPT4

**Comment by AkechiShiro** [comment]
@KilluaGremory you are correct, we should but I don't know if he will answer us or if he's gone MIA

**Comment by AkechiShiro** [comment]
If the license allows me to do everything I need then we don't need to wait for his answer @KilluaGremory

**Comment by AkechiShiro** [comment]
Just checked the license is GPL, I cannot do non open source stuff, but I can do open source fork, which is what I planned initially

**Comment by AkechiShiro** [comment]
@icpryde @Ibnelaiq here is the link of the Discord : https://discord.com/invite/k2JAqTfY

**Comment by AkechiShiro** [comment]
@Yink do I have permission to fork this repo and copy all the code and upgrade it ?

**Comment by Yink** [comment]
Hi, thanks for all the interest! I will not continue further on this, so yes please fork and upgrade it however you want :) Just link to this repo and give credit please.

**Comment by Bullicraft** [comment]
@AkechiShiro I would like to help in this project (but i'm not good at programming). @Yink may i ask how much time you took to create amadeus please?

**Comment by chovypilled** [comment]
I've been working on [rewriting it](https://github.com/chovypilled/Amadeus_kt) in Kotlin if anyone's interested. Never done anything like this before so it should be interesting lol
(edited to reflect the newer repo)

**Comment by icpryde** [comment]
@chovypilled thats awesome I’ll follow your progress, feel like using AI tools at this point just makes so much sense
